jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py

```diff
@@ -48,81 +52,110 @@ def __init__(
+
+    def _init_executors(self, dump_path, kwargs, startup_sync, startup_sync_args):
+        if startup_sync_args is None and startup_sync:
+            startup_sync_args = {}
+        # if the user passes args for syncing, they must've meant to sync as well
+        if startup_sync_args and (startup_sync is False or startup_sync is None):
+            startup_sync = True
+
+        if dump_path is None and startup_sync is None:
             name = getattr(self.metas, 'name', self.__class__.__name__)
             self.logger.warning(
                 f'No "dump_path" or "use_dump_func" provided '
                 f'for {name}. Use .rolling_update() to re-initialize...'
             )
-
-        if use_dump_func:
+        if startup_sync:
             self._kv_indexer = PostgreSQLStorage(**kwargs)
-            dump_func = self._kv_indexer.get_snapshot
-            self._vec_indexer = FaissSearcher(dump_func=dump_func, **kwargs)
+            self._vec_indexer = FaissSearcher(**kwargs)
+            self.sync(parameters=startup_sync_args, startup=True)
         else:
             self._kv_indexer = PostgreSQLStorage(**kwargs)
             self._vec_indexer = FaissSearcher(dump_path=dump_path, **kwargs)
 
     @requests(on='/sync')
-    def sync(self, parameters: Dict, **kwargs):
+    def sync(self, parameters: Optional[Dict], **kwargs):
         """
         Sync the data from the PSQLStorage into the FaissSearcher
         """
         use_delta = parameters.get('use_delta', False)
         only_delta = parameters.get('only_delta', False)
 
         if only_delta:
+            self._sync_only_delta(parameters, **kwargs)
+
+        else:
+            self._sync_snapshot(use_delta)
+
+    def _sync_snapshot(self, use_delta):
+        self.logger.info('Syncing via snapshot...')
+        if self.total_shards:
+            dump_func = functools.partial(
+                self._kv_indexer.get_snapshot, total_shards=self.total_shards
+            )
+            timestamp = self._kv_indexer.last_snapshot_timestamp
+            self._vec_indexer = FaissSearcher(dump_func=dump_func, **self._init_kwargs)
+
+            if use_delta:
+                self.logger.info(f'Now adding delta from timestamp {timestamp}')
+
+                deltas = self._kv_indexer._get_delta(
+                    shard_id=self.runtime_args.pea_id,
+                    total_shards=self.total_shards,
+                    timestamp=timestamp,
+                )
+                # deltas will be like DOC_ID, OPERATION, DATA
+                self._vec_indexer._add_delta(deltas)
+        else:
+            self.logger.warning(
+                'total_shards is None, rolling update '
+                'via PSQL import will not be possible.'
+            )
+
+    def _sync_only_delta(self, parameters, **kwargs):
+        timestamp = parameters.get('timestamp', None)
+        startup = kwargs.get('startup', False)
```

Comment:
I think this should be part of the parameters. If someone sends a request with `parameters={'startup': True}` it's their fault.